### ISSUE TITLE
The size of edge_Index becomes zero when using NeighborLoader

### ISSUE THREAD

**Issue by RX28666** [pyg-team/pytorch_geometric#5827]
### 🐛 Describe the bug

Hello, I am using NeighborLoader to do sampling on ogb-arxiv dataset (the total number of nodes=169343), when I set batch_size=169343, num_neighbors=[-1], everything is fine, but when I set batch_size=169343, num_neighbors=[25, 10], after several epochs, the bug appeared, the matrix of adj matrix becomes (169342, 169342), which is just one less than the number it should be(169343), then I set batch_size=32, num_neighbors=[-1], same bug appeared, the shape of edge_index becomes 0 this time, the error message is here:
addmm: Argument #3 (dense): Expected dim 0 size 0, got 32

My codes runs well on other datasets, like Cora, Citeseer

I print the size of that batch:
     batch.x.shape: torch.Size([32, 128])
     batch.edge_index.shape: torch.Size([2, 0])
     batch.edge_index: tensor([], device='cuda:0', size=(2, 0), dtype=torch.int64)
     adj.shape: (0, 0)

the following is my training function:

def train(data):
    model.train()
    for batch in train_loader:
        optimizer.zero_grad()
        adj_train = torch_geometric.utils.to_scipy_sparse_matrix(to_undirected(batch.edge_index))
        print("batch_train", batch.x.shape)
        print("edge_train", batch.edge_index.shape)
        print("edge_train", batch.edge_index)
        print("adj_train", adj_train.shape)
        train = train_mask[batch.n_id][:batch.batch_size]
        y = batch.y.squeeze(1)[:batch.batch_size][train]
        out = model(batch.x, adj_train, batch.n_id)[:batch.batch_size][train]
        loss = F.cross_entropy(out, y)
        loss.backward(retain_graph=True)
        optimizer.step()
    return float(loss)

And the following is my train_loader:
kwargs = {'batch_size': 32, 'num_workers': 2, 'persistent_workers': False}
train_loader = NeighborLoader(data, input_nodes=None,
                              num_neighbors=[25, 10], shuffle=True, **kwargs)

Datasets:
dataset = PygNodePropPredDataset(name='ogbn-arxiv')
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
data = dataset[0]
data = data.to(device)
split_idx = dataset.get_idx_split()
train_idx = split_idx['train'].to(device)
evaluator = Evaluator(name='ogbn-arxiv')
train_mask = torch.zeros(data.num_nodes, dtype=torch.bool)
train_mask[train_idx.long()] = True  #similar as planetoid

Thanks for your help!

### Environment

* PyG version:2.1.0
* PyTorch version: 1.12
* OS: google colab
* Python version:3
* CUDA/cuDNN version:11.3
* How you installed PyTorch and PyG (`conda`, `pip`, source): pip
* Any other relevant information (*e.g.*, version of `torch-scatter`):none

**Comment by rusty1s** [comment]
Thanks for reporting. Since we do no longer maintain the neighbor sampling of `torch-sparse` but have moved to a more optimized version in [`pyg-lib`](https://github.com/pyg-team/pyg-lib), can you do me a favor and test if the same issue appears there?
```
pip install pyg-lib -f https://data.pyg.org/whl/nightly/torch-1.12.0+${CUDA}.html
pip install git+https://github.com/pyg-team/pytorch_geometric.git
```

**Comment by RX28666** [comment]
Thanks so much for replying! It works and no errors are reported. Thanks again.

**Comment by RX28666** [comment]
Sorry to bother you again. I tried to use NeighborLoader on some datasets with isolated nodes, and then use `to_scipy_sparse_matrix` to covert edge_index to adj matrix, like following:
```
data = dataset[0]
data.edge_index = to_undirected(data.edge_index)
data = data.to(device)
data.n_id = torch.arange(data.num_nodes)
kwargs = {'batch_size': 3327, 'num_workers': 0, 'persistent_workers': False}
train_loader = NeighborLoader(data, input_nodes=None,
                              num_neighbors=[-1], shuffle=True, **kwargs)
for i in range(500):
    for batch in `train_loader:
        print(torch_geometric.utils.to_scipy_sparse_matrix(to_undirected(batch.edge_index)).shape)
```
I am using Citeseer, I put all 3327 nodes into one batch for verifying firstly, but the output of the above code not always 3327, sometimes becomes 3326, like following:
```
(3327, 3327)
(3327, 3327)
(3327, 3327)
(3327, 3327)
(3327, 3327)
(3326, 3326)
(3327, 3327)
(3327, 3327)
(3327, 3327)
```
I tried the method you mentioned before, but this problem still happened. 
I was wondering if If it's because of isolated nodes, but sometimes the result is correct(3327), sometimes is wrong(3326).
Any help would be appreciated.
